vyper/codegen/module.py

```diff
@@ -162,14 +172,16 @@ def _runtime_ir(runtime_functions, all_sigs, global_ctx):
 
     runtime = [
         "seq",
-        # check that calldatasize is at least 4, otherwise
-        # calldataload will load zeros (cf. yellow paper).
-        ["if", ["lt", "calldatasize", 4], ["goto", "fallback"]],
         ["with", "_calldata_method_id", shr(224, ["calldataload", 0]), selector_section],
         close_selector_section,
         ["label", "fallback", ["var_list"], fallback_ir],
     ]
 
+    if has_zero_selector_functions:
+        # check that calldatasize is at least 4, otherwise
+        # calldataload will load zeros (cf. yellow paper).
+        runtime.insert(1, ["if", ["lt", "calldatasize", 4], ["goto", "fallback"]])
```

Comment:
i think the flow of this will be clearer if it's more like
```
runtime = ["seq"]
if has_zero_selector_functions:
  runtime.append(["if", ...])
runtime.extend(
  ...
)
```

Reply:
oh, cool. thanks!

Reply:
changed with my latest commit